pywin32_testall.py

```diff
@@ -62,11 +62,22 @@ def find_and_run(possible_locations, script, cmdline_rest=""):
                            % (script, possible_locations))
 
 if __name__=='__main__':
+    import argparse
+
     code_directories = [this_dir] + site_packages
 
-    # win32
-    maybes = [os.path.join(directory, "win32", "test") for directory in code_directories]
-    find_and_run(maybes, 'testall.py')
+    parser = argparse.ArgumentParser(description="A script to trigger tests in all subprojects.")
+    parser.add_argument("-no-user-interaction",
+                        default=False,
+                        action='store_true',
+                        help="Configure the Python environment correctly for pywin32.")
+
+    args = parser.parse_args()
+
+    if not args.no_user_interaction:
+        # win32
```

Comment:
Note that most of the win32 test suite runs without interaction, and testall.py already has a list of tests that require interaction - so you probably want to pass the same cmdline arg to testall.py

Reply:
I took a look into testall and it seems that it should already skip these tests?
Don't see clearly the point where the switch should be added :confused:

Reply:
> Don't see clearly the point where the switch should be added

Yeah, me either :( Apart from directsound, what issues are you having?

Reply:
Well, at the moment all errors are displayed but ignored. I guess the next step will be to make the tests really fail in Appveyor.
So, if you accept, we can run all tests at least, which I think is a great step forward :)